proxy/http2/Http2ConnectionState.cc

```diff
@@ -1067,16 +1067,36 @@ Http2ConnectionState::find_stream(Http2StreamId id) const
 void
 Http2ConnectionState::restart_streams()
 {
-  Http2Stream *s = stream_list.head;
-
-  while (s) {
-    Http2Stream *next = static_cast<Http2Stream *>(s->link.next);
+  // x is a static variable, so it is shared in Http2ConnectionState instances and will get incremented in subsequent calls.
+  // It doesn't need to be initialized with rand() nor time(), and doesn't need to be accessed with a lock, because it doesn't need
+  // that randomness and accuracy.
+  static int x = 0;
+
+  Http2Stream *s   = stream_list.head;
+  Http2Stream *end = s;
+  if (s) {
+    // Change the start point randomly
+    for (int i = x % total_client_streams_count; i >= 0; --i) {
+      end = static_cast<Http2Stream *>(end->link.next ? end->link.next : stream_list.head);
+    }
+    s = static_cast<Http2Stream *>(end->link.next ? end->link.next : stream_list.head);
+
+    // Call send_response_body() for each streams
+    while (s != end) {
```

Comment:
It looks like if this loop is like below, we can remove the status check and calling `send_response_body()` outside of this loop.
```
while (s) {
  ...
  send_resopnse_body();
  ...
  if (s == end) {
    break;
  } else {
    s = next;
  }
}
```

Reply:
That makes loop condition unclear since `s` is always true in this loop. In my opinion, the check for stream condition should be moved into `send_response_body()`. Then calling one more `send_response_body()` after the loop wouldn't be a big deal, and it should be checked on every calls of `send_response_body()`. I'm not a big fun of breaking out from loops.

Reply:
All right, leave this. It's trivial.